client.go

```diff
@@ -191,8 +196,26 @@ func (c *client) getTransactionCount(ctx context.Context, account common.Address
 	return uint64(result), err
 }
 
+func (c *client) SendTransaction(ctx context.Context, tx *types.Transaction) (string, error) {
+	raw, err := rlp.EncodeToBytes(tx)
+	if err != nil {
+		return "", err
+	}
+	return c.SendRawTransaction2(ctx, raw)
+}
+
 func (c *client) SendRawTransaction(ctx context.Context, tx []byte) error {
-	return c.r.CallContext(ctx, nil, "eth_sendRawTransaction", common.ToHex(tx))
+	_, err := c.SendRawTransaction2(ctx, tx)
+	return err
+}
+
+func (c *client) SendRawTransaction2(ctx context.Context, tx []byte) (string, error) {
+	var result hexutil.Bytes
+	err := c.r.CallContext(ctx, &result, "eth_sendRawTransaction", common.ToHex(tx))
+	if err != nil {
+		return "", err
+	}
+	return result.String(), nil
```

Comment:
Are these leftover from debugging?

Reply:
No, I actually want the tx hash returned. Not sure why the original ones didn't return useful information. Had to add new function to not break the API.

Reply:
The transaction hash is `keccak256(tx)`. There is no need to retrieve it from the rpc call.

Reply:
is that good UX?

Reply:
IMHO yes. It is trivial. And if you produced the bytes from a types.Transaction, then you already have it on hand.

Reply:
re: breaking the api - we're still pre-1.0, so we can just bump the minor version if we need to.